### PR TITLE
WIP: feat: add support for desktop notifications

### PR DESCRIPTION
**Pull request by manekinekko**
fixes https://github.com/nrwl/angular-console/issues/164

### CODE REVIEW

**Comment by manekinekko** [comment]
@vsavkin @mrmeku can you help explain how the settings are handled by the server?

File: libs/utils/src/lib/notification.service.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export class Notification {
+  constructor(private readonly settings: Settings) {}
+
+  notify(body: string, title: string = 'Angular Console'): Observable<any> {
+
+    if (!this.settings.enableNotifications) {
+        return of({});
+    }
+
+    const notificationInstance = (window as any).Notification.requestPermission().then(
```
Comment (mrmeku) [inline libs/utils/src/lib/notification.service.ts]: I think `@types/chrome` will have the typings for this api

**Comment by mrmeku** [comment]
@manekinekko Here's how the server deals with settings: https://github.com/nrwl/angular-console/blob/master/server/src/api/read-settings.ts#L3 

That class has both a read and write method attached to it. 

Here is the api endpoint which the front end calls to update a user's preferences: 
https://github.com/nrwl/angular-console/blob/a9f4926b808fbf592286b875fea32a6f42245591/server/src/schema/resolvers.ts#L441

**Comment by mrmeku** [comment]
I'm going to close this PR because it pretty stale now. Please re-open it when you get a chance to rebase!

**Pull request closed without merging**